File: src/main/java/bad/robot/radiate/Environment.java
package bad.robot.radiate;

import static java.lang.String.format;

public class Environment {

    public static String getEnvironmentVariable(String variable) {
        if (System.getenv(variable) == null)
            throw new IllegalArgumentException(format("Please set environment variable '%s'", variable));
        return System.getenv(variable);
    }

    public static String getEnvironmentVariable(String variable, String defaultValue) {
        if (System.getenv(variable) == null)
            return defaultValue;
        return System.getenv(variable);
    }
}

File: src/main/scala/bad/robot/radiate/Environment.scala
package bad.robot.radiate

import bad.robot.radiate.OptionSyntax._

import scalaz.syntax.std.option._

object Environment {

  def getEnvironmentVariable(variable: String): Option[String] = {
    NonEmptyOption(sys.env.get(variable))
  }

  def getEnvironmentVariable(variable: String, defaultValue: String): String = {
    NonEmptyOption(sys.env.get(variable)).some(identity).none(defaultValue)
  }
}
